### PR TITLE
[AOT Refactor] dispatch compile graph

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114643
* #114564
* __->__ #114563
* #114562
* #114561
* #114559
* #114558
* #114557
* #114556
* #114555
* #114554
* #114553
* #114552
* #114551
* #114550
---

Part _ of https://github.com/pytorch/pytorch/issues/114548

### CODE REVIEW

File: torch/_higher_order_ops/map.py [hunk 1]
```diff
@@ -133,7 +137,7 @@ def fw_with_masks(*args):
                     for ret in fw_out
                 ]
 
-            joint = create_joint(fw_with_masks, aot_config=dummy_aot_config)
+            joint = create_joint_function(fw_with_masks, aot_config=dummy_aot_config)
```
[the remark this answers is not in the document]
Comment (jon-chuang) [reply torch/_higher_order_ops/map.py:140]: Same as from/to/is_functional

**Review by bdhirsh — APPROVED** [review]

**Pull request closed without merging**